Trimble.Modus.Components/Controls/TMModal/TMModalContents.xaml

```diff
@@ -0,0 +1,114 @@
+    </ResourceDictionary>
+  </ContentPage.Resources>
+  <!-- Note: Using this parent container to create shadow effect for the modal since we are unable to add shadow to the modal/ModalContainer directly.
+             Change the background color of the parent container to change the color of the shadow.
+             Change the padding of the parent container to change the size of the shadow -->
+  <Border x:Name="ParentContainer"
+          StrokeThickness="0"
+          Padding="3"
+          BackgroundColor="{x:StaticResource ModalShadowColor}"
+          VerticalOptions="Center"
+          HorizontalOptions="Center"
+          StrokeShape="RoundRectangle 5">
+    <Border x:Name="ModalContainer"
+            Padding="16"
+            VerticalOptions="Center"
+            HorizontalOptions="Fill"
+            StrokeShape="RoundRectangle 4"
+            Stroke="{x:StaticResource Black}"
+            StrokeThickness="0"
+            BackgroundColor="White">
+      <Border.Shadow>
+        <Shadow Brush="{x:StaticResource Black}"
+                Offset="0,0"
+                Opacity="0.2"
+                Radius="8" />
+      </Border.Shadow>
+      <Grid RowSpacing="16">
+        <Grid.ColumnDefinitions>
+          <ColumnDefinition Width="*" />
+          <ColumnDefinition Width="*" />
+          <ColumnDefinition Width="*" />
+        </Grid.ColumnDefinitions>
+        <Grid.RowDefinitions>
+          <RowDefinition Height="Auto" />
+          <RowDefinition Height="*" />
+          <RowDefinition Height="Auto" />
+        </Grid.RowDefinitions>
+        <StackLayout  Grid.Column="0"
+                      Grid.Row="0"
+                      Orientation="Horizontal">
+          <Image x:Name="IconImage"
+               Source="{Binding TitleIcon}"
+               IsVisible="{Binding TitleIcon,Converter={StaticResource ImageNullOrEmptyBoolConverter}}"
+               VerticalOptions="Center"
+               HeightRequest="26"
+               WidthRequest="26"
+               Margin="0,0,8,0"/>
+          <Label x:Name="TitleLabel"
+               Text="{Binding Title}"
+               IsVisible="{Binding Title,Converter={StaticResource StringNotNullOrEmptyBoolConverter}}"
+               VerticalOptions="Center"
+               FontFamily="OpenSans-Semibold"
+               FontSize="16"
+               TextColor="{x:StaticResource Black}"
+               Padding="0,0,8,0" />
+        </StackLayout>
+        
+        <Image Source="{x:Static constants:ImageConstants.CloseButtonImage}"
+               Grid.Column="2"
+               Grid.Row="0"
+               VerticalOptions="Center"
+               HeightRequest="16"
+               WidthRequest="16"
+               Margin="2"
+               BackgroundColor="Transparent"
+               HorizontalOptions="End">
+          <Image.GestureRecognizers>
+            <TapGestureRecognizer Tapped="CloseModal" />
```

Comment:
This method should be exposed to user end. If they want to add some functionality when closing the popup

Reply:
No need Sofiya. The user has to know when the modal is closing - via a listener, but I think we already have that. @BAL4-R to confirm.

Reply:
Done